spring-integration-file/src/main/java/org/springframework/integration/file/locking/FileChannelCache.java

```diff
@@ -43,13 +45,18 @@ private FileChannelCache() {
 
 	/**
 	 * Try to get a lock for this file while guaranteeing that the same channel will be used for all file locks in this
-	 * VM. If the lock could not be acquired this method will return <code>null</code>.
+	 * VM. If the lock could not be acquired this method will return <code>null</code>. If the file does not exist this
+	 * method will throw a {@link FileNotFoundException}.
 	 * <p>
 	 * Locks acquired through this method should be passed back to #closeChannelFor to prevent memory leaks.
 	 * <p>
 	 * Thread safe.
 	 */
 	public static FileLock tryLockFor(File fileToLock) throws IOException {
+		if (!fileToLock.exists()) {
+			throw new FileNotFoundException("Unable to get lock for non-existent file: " + fileToLock.getAbsolutePath());
```

Comment:
Wouldn't return `null` instead of this exception enough for the method contract?
As you see in the `NioFileLocker`, you just suppress such an exception any way.

So, let's make a `return null;` here, add a `@Nullable` for method return and so on!

Reply:
That's what I suggested as an alternative in #3076. In my mind, the semantics were different between _a lock that failed because the file was already locked (or any other filesystem-related reason)_, and _an attempt to lock a file that does not exist_.

Anyway, I will push a new version returning `null` 😉

Reply:
Well, if there is nothing to lock, then we definitely not locked. Seems for me that's the point of that contract.
So, let's live with `null`  for non-existing file as well!